libcnb-cargo/Cargo.toml

```diff
@@ -17,7 +17,10 @@ path = "src/main.rs"
 
 [dependencies]
 cargo_metadata = "0.14.1"
-clap = "2.34.0"
+clap = { version = "3.1.0", default-features = false, features = [
```

Comment:
Curious: Which default features do we not want here?

Reply:
Suggestions + colour:
https://docs.rs/clap/latest/clap/#feature-flags
https://github.com/clap-rs/clap/blob/v3.1.1/Cargo.toml#L58-L70

Neither were enabled by default with clap v2, so we previously didn't have them.

Since the cargo plugin is pretty simple, I don't think we need them (and would prefer to err on the side of fewer deps; death by 1000 cuts etc).